sway/tree/workspace.c

```diff
@@ -516,3 +517,114 @@ struct sway_container *workspace_output_get_highest_available(
 
 	return NULL;
 }
+
+struct pid_workspace {
+	pid_t pid;
+	char *workspace;
+	struct timespec time_added;
+
+	struct sway_container *output;
+	struct wl_listener output_destroy;
+
+	struct wl_list link;
+};
+
+static struct wl_list pid_workspaces;
```

Comment:
Can't this be in `sway_root`, and initialized there?

Reply:
Does it need to be? The pid namespace is inherently global on Linux, so it's not like we're gaining some feel-good ability to have two sway_roots or anything. I'd rather isolate it to the code that uses it.

Reply:
Yeah but this is a little meh:

```c
	if (!pid_workspaces.prev && !pid_workspaces.next) {
		wl_list_init(&pid_workspaces);
		return NULL;
	}
```

But I guess since this is not wlroots it doesn't matter too much.